### PR TITLE
ISPN-7351 Additional testcase for @Named qualifiers

### PR DESCRIPTION
**Pull request by slaskawi**
https://issues.jboss.org/browse/ISPN-7351

The implementation seems to be fine, adding only test cases.

### CODE REVIEW

**Comment by SetoKaiba** [comment]
@Named("large")
   @ConfigureCache("large")
   @Produces
   public Configuration largeConfiguration = new ConfigurationBuilder().memory().size(3).build();

   @Named("small")
   @ConfigureCache("large")
   @Produces
   public Configuration largeConfiguration = new ConfigurationBuilder().memory().size(3).build();

And then you will see the exception.

**Comment by SetoKaiba** [comment]
@Named qualifier is not treated as normal qualifier. Then the @Default will be automatically added. Then it will cause the exception.

**Comment by slaskawi** [comment]
Interesting! Let me check it!

File: cdi/embedded/src/test/java/org/infinispan/cdi/embedded/test/cachemanager/programmatic/Config.java [hunk 1]
```diff
@@ -18,6 +19,29 @@
...
+   public Configuration largeConfiguration = new ConfigurationBuilder().memory().size(10).build();
+
+   /**
+    * The same as the above. The intention here is to check whether we can use 2 Configurations with <code>@Named</code>
+    * annotations.
+    */
+   @Named("super-large")
+   @ConfigureCache("super-large")
+   @Produces
+   public Configuration superLargeConfiguration = new ConfigurationBuilder().memory().size(20).build();
```
Comment (slaskawi) [inline cdi/embedded/src/test/java/org/infinispan/cdi/embedded/test/cachemanager/programmatic/Config.java]: I added another Configuration which uses `@Named` annotation. It seems fine on my machine, but let's check if it's green on [TeamCity](http://ci.infinispan.org).

Comment (SetoKaiba) [inline cdi/embedded/src/test/java/org/infinispan/cdi/embedded/test/cachemanager/programmatic/Config.java]: The exception only exists in 8.x. You're right. It's working correctly with 9.x.

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**